src/main.rs

```diff
@@ -118,7 +131,20 @@ fn settings() -> io::Result<Settings> {
         },
     )?;
 
-    Ok(Settings { threshold })
+    let keep = match matches.values_of(KEEP_OPTION) {
+        Some(values) => match RegexSet::new(values) {
+            Ok(set) => Some(set),
+            Err(_) => {
+                return Err(io::Error::new(
+                    io::ErrorKind::InvalidInput,
```

Comment:
While we're here, would you mind changing line 127 to `io::ErrorKind::InvalidInput` as well? I think it's a better choice than what I had there (`io::ErrorKind::Other`).